core/src/main/java/org/apache/iceberg/util/ContentFileUtil.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.util;
+
+import java.util.Set;
+import org.apache.iceberg.ContentFile;
+
+public class ContentFileUtil {
+  private ContentFileUtil() {}
+
+  /**
+   * Copies the {@link ContentFile} with the specific stat settings.
+   *
+   * @param file a generic data file to copy.
+   * @param withStats whether to keep any stats
+   * @param columnsToKeepStats a set of column ids to keep stats. If empty or <code>null</code> then
+   *     every column stat is kept.
```

Comment:
This should not keep all stats when empty.

Reply:
I find this logic a bit hard to follow cause we control the behavior using two variables. Any way we can simplify that? We could have something like below.

```
Scan.java

includeColumnStats() -> include all stats
includeColumnStats(Collection<String> requestedColumns) -> include only stats for requested columns

ContentFile.java

copy() -> copy preserving all loaded stats (existing)
copyWithoutStats() -> copy without any loaded stats (existing)
copy(boolean withStats) -> calls either copy() or copyWithoutStats() (existing)
copyWithStats(Collection<String> requestedColumns) -> copy with stats only for requested columns
```

We can then maintain a list of columns to project stats in our scans or in the scan context object.
- A collection with a single element `*` means we will call copy() on files.
- Null or empty collection means we will call copyWithoutStats() on files.
- A collection with column names means we will call copyWithStats() on files.

All of that logic above can be incapsulated in a single copy method in our base scan.

Reply:
Any thoughts, @pvary @rdblue @stevenzwu?

Reply:
@aokolnychyi  the proposed interfaces for `Scan` and `ContentFile` make sense to me. 

> A collection with a single element * means we will call copy() on files.
> Null or empty collection means we will call copyWithoutStats() on files.

Regarding the collection of a single `*` element, it feels a little tacky to me. The util method/class takes the  two configs from the `TableScanContext` and implements the copy logic. 

> All of the logic above can be incapsulated in a single copy method in our base scan.

This `ContentFileUtil` was intended for that purpose of incapsulating all those logic. Note that `ManifestGroup` class also leverages this util method.

Reply:
I converted the `Scan` to use the column names instead of the column ids, like: `Scan.includeColumnStats(Collection<String> requestedColumns)`

For the `ContentFile` we do not have the `Schema` at hand, so I had to stick to the `ContentFile.copyWithStats(Set<Integer> requestedColumnIds)`

Reply:
@rdblue, @aokolnychyi, @stevenzwu: I would appreciate your thoughts here.
- Do you have any better idea how to handle `ContentFile.copyWithStats` other than keeping them as Ids?
- Is it ok, to keep it as a `Set`, or shall I turn this to a `Collection` as well?